src/rpc/mempool.cpp

```diff
@@ -587,6 +588,99 @@ static RPCHelpMan getmempoolentry()
     };
 }
 
+static RPCHelpMan gettxspendingprevout()
+{
+    return RPCHelpMan{"gettxspendingprevout",
+        "Scans the mempool to find transactions spending any of the given outputs",
+        {
+            {"outputs", RPCArg::Type::ARR, RPCArg::Optional::NO, "The transaction outputs that we want to check, and within each, the txid (string) vout (numeric).",
+                {
+                    {"", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED, "",
+                        {
+                            {"txid", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The transaction id"},
+                            {"vout", RPCArg::Type::NUM, RPCArg::Optional::NO, "The output number"},
+                        },
+                    },
+                },
+            },
+        },
+        RPCResult{
+            RPCResult::Type::ARR, "", "",
+            {
+                {RPCResult::Type::OBJ, "", "",
+                {
+                    {RPCResult::Type::STR_HEX, "txid", "the transaction id of the spent output"},
+                    {RPCResult::Type::NUM, "vout", "the vout value of the spent output"},
+                    {RPCResult::Type::STR_HEX, "spendingtxid", /*optional=*/true, "the transaction id of the mempool transaction spending this output (omitted if unspent)"},
```

Comment:
This message needs to be changed, as the transaction may not be in the mempool, but in the blocks.
```suggestion
                    {RPCResult::Type::STR_HEX, "spendingtxid", /*optional=*/true, "the id of the transaction that spends the output"},
```

Reply:
done in https://github.com/bitcoin/bitcoin/commit/84c1e67c7b6a7fa618325e2767b243f97cab0e45